### PR TITLE
[Torch] Add support for max_pool1d

### PR DESCRIPTION
**Pull request by wyc-ruiker**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

add max_pool1d in https://github.com/apache/incubator-tvm/issues/5133, @masahi @alexwong

### CODE REVIEW

File: tests/python/frontend/pytorch/test_forward.py [hunk 1]
```diff
@@ -363,9 +363,35 @@ class MaxPool2D2(Module):
...
+            return torch.nn.MaxPool1d(kernel_size=10)(args[0])
+
+    class MaxPool1D3(Module):
+        def forward(self, *args):
+            return torch.nn.MaxPool1d(kernel_size=4, padding=2, stride=2)(args[0])
+
+    input_data = torch.rand(input_shape).float()
+    verify_model(MaxPool1D1().float().eval(), input_data=input_data)
+    verify_model(MaxPool1D2().float().eval(), input_data=input_data)
+    verify_model(MaxPool1D3().float().eval(), input_data=input_data)
```
Comment (masahi) [inline tests/python/frontend/pytorch/test_forward.py]: No need to write a wrapper class. See https://github.com/apache/incubator-tvm/blob/86079479f0556002adfce2f438ea2a607e318c23/tests/python/frontend/pytorch/test_forward.py#L704-L732

File: python/tvm/relay/frontend/pytorch.py [hunk 2]
```diff
@@ -213,12 +213,33 @@ def _impl(inputs, input_types):
         pool_size = _infer_shape(inputs[1])
         strides = _infer_shape(inputs[2])
         padding = _infer_shape(inputs[3])
-
+        dilation = _infer_shape(inputs[4])
```
Comment (masahi) [inline python/tvm/relay/frontend/pytorch.py]: does pooling has dilation argument?

File: src/relay/op/nn/pooling.cc [hunk 3]
```diff
@@ -987,6 +987,10 @@ Array<te::Tensor> Pool1DCompute(const Attrs& attrs,
       << " or 4-D input (e.g. NCWc on for vector instructions)"
       << " or 5-D input (e.g. NCWnc for tensor accelerators)";
 
+  if (param->padding.size() == 1) {
+    padding.push_back(padding[0]);
```
Comment (masahi) [inline src/relay/op/nn/pooling.cc]: Why do we need this? Does 1D pooling require two pad values (left & right)?

File: tests/python/frontend/pytorch/test_forward.py [hunk 4]
```diff
@@ -363,9 +363,35 @@ class MaxPool2D2(Module):
         def forward(self, *args):
             return torch.nn.MaxPool2d(kernel_size=[10, 10])(args[0])
 
+    class MaxPool2D3(Module):
```
Comment (masahi) [inline tests/python/frontend/pytorch/test_forward.py]: Remove this wrapper

File: src/relay/op/nn/pooling.cc (hunk 3 above)
Comment (wyc-ruiker) [reply src/relay/op/nn/pooling.cc]: Yes, If we don't have this, the max_pool1d will fail in https://github.com/apache/incubator-tvm/blob/7c5ff50873e91e9ad27b5f08847c27d58e8b5c4c/topi/include/topi/nn/pooling.h#L679-L680

File: python/tvm/relay/frontend/pytorch.py (hunk 2 above)
Comment (wyc-ruiker) [reply python/tvm/relay/frontend/pytorch.py]: In https://pytorch.org/docs/stable/nn.html#maxpool1d, MaxPool1d and MaxPool2d have dilation argument.

File: tests/python/frontend/pytorch/test_forward.py (hunk 4 above)
Comment (wyc-ruiker) [reply tests/python/frontend/pytorch/test_forward.py]: Don't we need a test for padding and stride in Maxpool?

Comment (wyc-ruiker) [reply tests/python/frontend/pytorch/test_forward.py]: Don't we need a test for padding and stride in Maxpool?

Comment (masahi) [reply tests/python/frontend/pytorch/test_forward.py]: I am talking about the same point as https://github.com/apache/incubator-tvm/pull/5142/files#r397046667

Yes, you should have a test, but no need to write a wrapper class

Comment (wyc-ruiker) [reply tests/python/frontend/pytorch/test_forward.py]: > I am talking about the same point as https://github.com/apache/incubator-tvm/pull/5142/files#r397046667
> 
> Yes, you should have a test, but no need to write a wrapper class

Thanks. I misunderstood your review suggestion :-)

File: tests/python/frontend/pytorch/test_forward.py (hunk 1 above)
Comment (wyc-ruiker) [reply tests/python/frontend/pytorch/test_forward.py]: done.

File: tests/python/frontend/pytorch/test_forward.py (hunk 4 above)
Comment (wyc-ruiker) [reply tests/python/frontend/pytorch/test_forward.py]: done.

**Comment by masahi** [comment]
Thanks @wyc-ruiker
